pkg/cache/cache.go

```diff
@@ -128,6 +128,14 @@ type Options struct {
 	// Be very careful with this, when enabled you must DeepCopy any object before mutating it,
 	// otherwise you will mutate the object in the cache.
 	UnsafeDisableDeepCopyByObject DisableDeepCopyByObject
+
+	// TransformFuncByObject is a map from GVKs to transformer functions which
+	// get applied when objects of the transformation are about to be committed
+	// to cache.
+	//
+	// This function is called both for new objects to enter the cache,
+	// 	and for updated objects.
+	TransformFuncByObject TransformFuncByObject
```

Comment:
Would it make sense to also add a `DefaultTransform` that is used if there is not a more specific one?

Reply:
That sounds potentially very useful to some cases. I will add that in.

Reply:
> Would it make sense to also add a `DefaultTransform` that is used if there is not a more specific one?

I'm not sure if we really need the `DefaultTransform`. This is different with the object selector IMHO.
There might be the same  `LabelSelector` for watching multiple GVKs, but since different kinds of object have different spec/status structure, they should be unlikely to have the same `Transform` logic.

Reply:
I am in agreement with Alvaro. My first instinct was that one might want to use a global transform to strip fields from ALL objects. Additionally, it may be useful to write a transformer that modifies all objects conforming to a given interface. A `DefaultTransformer` would allow that.

Reply:
Default transformer & test has been added. Marking as resolved.